pkg/compute/executor.go

```diff
@@ -167,10 +167,18 @@ func (e *BaseExecutor) Run(ctx context.Context, localExecutionState store.LocalE
 		}
 	}()
 
+	// We expect the current state to be BidAccepted but in some cases we will be restarting
+	// a failed job which will be in the Cancelled state (although this may later be upgraded
+	// to a FailedState).
+	expectedState := store.ExecutionStateBidAccepted
+	if localExecutionState.State == store.ExecutionStateCancelled {
+		expectedState = store.ExecutionStateCancelled
+	}
+
```

Comment:
I don't get this. If the execution is in cancelled state, then we shouldn't run or retry it

Reply:
This is why I'm hoping for a Failed state.  At the moment, it is us who cancelled the execution in startup.go